src/core/connection-parameters.cpp

```diff
@@ -119,6 +119,16 @@ connection_parameters::connection_parameters(connection_parameters const& other)
         backendRef_->inc_ref();
 }
 
+connection_parameters::connection_parameters(connection_parameters && other)
+    : factory_(std::move(other.factory_)),
+      connectString_(std::move(other.connectString_)),
+      backendRef_(std::move(other.backendRef_)),
+      options_(std::move(other.options_))
+{
+    other.factory_ = nullptr;
```

Comment:
IMO it's easier to read/follow -- especially for occasional trespassers like me. It avoids the question "why is some raw-pointer nulled out and the other?". But as you like, let me know.

Regarding movable logger, I've tried but gave up. The problem is the way it is constructed, with the strict assumption that 'm_impl' member is never null. It even throws an exception when constructed with logger_impl = nullptr. Making it movable would result in such a case. It looks hypothetical, but I don't know about the details. So left it alone.

Now looking around how the logger is used etc, I think there are some more places which would need touch-up if logger was made movable. It also looks like it could use std::unique_ptr instead of a raw pointer here to make things easier. But I'd rather make this a follow-up commit / PR if you don't mind.

Reply:
Sure, let's postpone making `logger` moveable for another PR, I didn't realize it was non-trivial, sorry.